modulemd/v2/include/modulemd-2.0/private/modulemd-yaml.h

```diff
@@ -421,4 +421,129 @@ modulemd_yaml_emit_document_headers (yaml_emitter_t *emitter,
+    {                                                                         \
+      if (value != NULL)                                                      \
+        {                                                                     \
+          EMIT_KEY (emitter, error, key);                                     \
+          if (!mmd_emitter_scalar (                                           \
+                emitter, value, YAML_PLAIN_SCALAR_STYLE, error))              \
+            return FALSE;                                                     \
+        }                                                                     \
+    }                                                                         \
+  while (0)
+
+#define EMIT_MAPPING_START(emitter, error)                                    \
+  do                                                                          \
+    {                                                                         \
+      if (!mmd_emitter_start_mapping (                                        \
+            emitter, YAML_BLOCK_MAPPING_STYLE, error))                        \
+        return FALSE;                                                         \
+    }                                                                         \
+  while (0)
+
+#define EMIT_MAPPING_END(emitter, error)                                      \
+  do                                                                          \
+    {                                                                         \
+      if (!mmd_emitter_end_mapping (emitter, error))                          \
+        return FALSE;                                                         \
+    }                                                                         \
+  while (0)
+
+#define EMIT_SEQUENCE_START(emitter, error)                                   \
+  do                                                                          \
+    {                                                                         \
+      if (!mmd_emitter_start_sequence (                                       \
+            emitter, YAML_BLOCK_SEQUENCE_STYLE, error))                       \
+        return FALSE;                                                         \
+    }                                                                         \
+  while (0)
+
+#define EMIT_SEQUENCE_END(emitter, error)                                     \
+  do                                                                          \
+    {                                                                         \
+      if (!mmd_emitter_end_sequence (emitter, error))                         \
+        return FALSE;                                                         \
+    }                                                                         \
+  while (0)
+
+#define EMIT_HASHTABLE_VALUES(emitter, error, key, table, emitfn)             \
```

Comment:
This function checks for presence but it differs in name from `EMIT_SCALAR_IF_SET()`. We should pick one convention and stick with it so it reads cleanly in the actual emit functions.